### PR TITLE
Redirect to the translated page with toast

### PR DESCRIPTION
**Pull request by pklion**
Automatically redirect to the Japanese page when the locale is ja. (#49)

### CODE REVIEW

**Comment by pklion** [comment]
But the language select link won't work properly...

File: index.html [hunk 1]
```diff
@@ -155,6 +155,11 @@ <h2 class="fw-bold">Take a look inside</h2>
             y.parentNode.insertBefore(t, y);
           })(window, document, "clarity", "script", "4q1wajdktz");
         </script>
+        <script>
+          if (window.navigator.language.slice(0, 2) == 'ja'){
+            location.href = 'ja-jp/index.html';
+          }
+        </script>
```
Comment (chingucoding) [inline index.html:162]: If someone with locale japan want's to view the en file, you immediately send them back. How about showing a small toast at the bottom suggesting to switch to ja?

Comment (chingucoding) [reply index.html:162]: Additional option: landing page is english, also introduce en-us/index.html which is an exact copy that can be used for language switching?

Comment (pklion) [reply index.html:162]: > If someone with locale japan want's to view the en file, you immediately send them back. How about showing a small toast at the bottom suggesting to switch to ja?
I think this is a great idea!
Could you make a toast design for that?

**Comment by chingucoding** [comment]
*Continuing here because of screenshots @pklion*

Created a small toast design on the index.html file on this branch: https://github.com/chingucoding/files-community.github.io/tree/toast-experiments

Currently, you can place it at the bottom or top right, not sure what the best position is here.

Screenshots:
Screenshot of landing page showing toast at the bottom of the image explaining that there is a Japanese translation
Screenshot of landing page showing toast in the top right corner of the image explaining that there is a Japanese translation

**Comment by pklion** [comment]
> _Continuing here because of screenshots @pklion_
> 
> Created a small toast design on the index.html file on this branch: https://github.com/chingucoding/files-community.github.io/tree/toast-experiments
> 
> Currently, you can place it at the bottom or top right, not sure what the best position is here.
> 
> Screenshots:
> Screenshot of landing page showing toast at the bottom of the image explaining that there is a Japanese translation
> Screenshot of landing page showing toast in the top right corner of the image explaining that there is a Japanese translation

Maybe using toasts of Bootstrap is also good.
I'm making a sample so please wait...

**Comment by chingucoding** [comment]
I wasn't using the bootstrap toast since there were talks about removing Bootstrap (and I was thinking about doing that some time soonish), so adding more dependencies on that wasn't something good right now.

**Comment by pklion** [comment]
@chingucoding 
OK, Please let me use yours.

**Comment by chingucoding** [comment]
@pklion Sure feel free to use it as you need! I think we might want to polish the design a bit more, maybe someone else has an opinion on this? 

In the meantime, either way, we would need a way to determine whether to show the toast or not, would be great if you can add that.

**Comment by pklion** [comment]
I'm a little working on updating the design (when the button hovered).
image

**Comment by chingucoding** [comment]
@pklion I pushed a few small changes to my toast branch regarding accessibility, just so you know. We should ensure that the landing page is accessible and usable for everyone.

**Comment by chingucoding** [comment]
@pklion Just to sync with you and prevent possible merge conflicts, would you like me to wait with #55 until this PR is merged?

**Comment by pklion** [comment]
@chingucoding Sorry for the late, and yes, please.

**Comment by pklion** [comment]
I also added the toast on the Japanese page.

**Comment by pklion** [comment]
It'll better if we can remember the state of the toast...

**Comment by chingucoding** [comment]
> Could I make this PR ready for review?

Sure thing, if you think it's ready for review, of course!

> It'll better if we can remember the state of the toast...

Yes agreed. I think best way is to use the [WebStorage API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API), namely `localStorage` and *show* the toast if we haven't stored that the dialogue was dismissed.

**Comment by pklion** [comment]
> Yes agreed. I think best way is to use the WebStorage API, namely localStorage and show the toast if we haven't stored that the dialogue was dismissed.
Could anyone make this? I'm not good at writing JS...

**Comment by yaichenbaum** [comment]
@pklion What about using cookies?

**Comment by chingucoding** [comment]
> @pklion What about using cookies?

@yaichenbaum We shoudn't use cookies if we have other options, especially since cookies will introduce legal issues (cookie banner etc). Mozilla recommends using Web Storage (https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Client-side_web_APIs/Client-side_storage)

**Comment by chingucoding** [comment]
@pklion I've added the necessary changes to the following branch: https://github.com/chingucoding/files-community.github.io/tree/toast-storage-behavior

The behavior is as followed: Show the toast, if it is dismissed on ANY page, we will never show it again.

**Comment by pklion** [comment]
> @pklion I've added the necessary changes to the following branch: https://github.com/chingucoding/files-community.github.io/tree/toast-storage-behavior
> 
> The behavior is as followed: Show the toast, if it is dismissed on ANY page, we will never show it again.

Thank you very much!

File: index.html [hunk 2]
```diff
@@ -24,25 +23,25 @@
...
+            -------------------------------=+*####
+            -----------------------------=+++*####
+            --------------------------=+++++++####
+            ------------------------=+++++++==####
+            --------------------==+++++++==--=####
+            ------------------=++++++++=-----=####
+            ---------------==+++++++==-------=####
+            ------------==+++++++++++++++++++++++
+            Files        +++++++++++++++++++++++++
+      -->
```
Comment (chingucoding) [inline index.html:44]: Please revert those formatting changes.

File: ja-jp/index.html [hunk 3]
```diff
@@ -60,7 +59,7 @@
                 </a>
               </div>
               <div class="col-2 align-self-center">
-                <a href="../index.html" class="text-white">
+                <a href="/" class="text-white">
```
Comment (chingucoding) [inline ja-jp/index.html:62]: Please use relative paths, makes it easier to develop/debug.

**Review by chingucoding — COMMENTED** [review]
@pklion Thank you for all your time in this! I've created a PR on your repository to get the latest changes to only show the toast if the toasts language is actually relevant or not.

File: ja-jp/index.html (hunk 3 above)
Comment (pklion) [reply ja-jp/index.html:62]: Already fixed in my repo :)
https://github.com/pklion/files-community.github.io/blob/translation/ja-jp/index.html

**Comment by pklion** [comment]
@chingucoding Thank you for your PR!

File: ja-jp/index.html (hunk 3 above)
Comment (chingucoding) [reply ja-jp/index.html:62]:
```suggestion
                <a href="../" class="text-white">
```

File: index.html [hunk 4]
```diff
@@ -60,17 +59,19 @@
                 </a>
               </div>
               <div class="col-2 align-self-center">
-                <a href="ja-jp/index.html" class="text-white">
+                <a href="ja-jp/" class="text-white">
```
Comment (chingucoding) [inline index.html:62]:
```suggestion
                <a href="./ja-jp/" class="text-white">
```

**Review by chingucoding — COMMENTED** [review]
Only minor things about paths, otherwise this is good to go! Thank you for your patience with this PR @pklion !

**Comment by pklion** [comment]
Done! @chingucoding Thank you for your kindness :-D

**Review by chingucoding — APPROVED** [review]
Looks good to me, thank you @pklion !!!
@yaichenbaum unless you have any objections, I'll merge this.

**Review by yaichenbaum — APPROVED** [review]

**Pull request merged**